test/ServerTest.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright 2024, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Julian Mundhahs (mundhahj@tf.uni-freiburg.de)
+
+#include <engine/Server.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <boost/beast/http.hpp>
+
+#include "util/GTestHelpers.h"
+#include "util/http/HttpUtils.h"
+#include "util/http/UrlParser.h"
+
+namespace http = boost::beast::http;
+using namespace ad_utility;
+
+TEST(ServerTest, parseHttpRequest) {
+  auto ParsedRequest = [](const std::string& path,
+                          const HashMap<std::string, std::string>& parameters,
+                          const std::optional<std::string>& query)
+      -> testing::Matcher<const UrlParser::ParsedRequest> {
+    return testing::AllOf(
+        AD_FIELD(UrlParser::ParsedRequest, ParsedRequest::path_,
```

Comment:
```suggestion
        AD_FIELD(UrlParser::ParsedRequest, path_,
```
Again (see below) , not sure why your syntax even works:)